src/components/UserInfo/UserInfo.tsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { User } from '../../typedefs/User';
+import '../PostList/PostList.scss';
+
+interface Props {
+  userInfo: Pick<User, 'email' | 'name'> | undefined,
```

Comment:
Then you wouldn't need this undefined type here
```suggestion
  userInfo: Pick<User, 'email' | 'name'>,
```